spec/factories/course_enrichments.rb

```diff
@@ -16,8 +16,8 @@
 
 FactoryBot.define do
   factory :course_enrichment do
-    provider
     course
+    provider { course.provider }
```

Comment:
I don't think `provider` & `course` are actually part of `course_enrichment`

relation is defined by `provider_code` & `ucas_course_code`

It be nice to remove

Reply:
What do you mean @defong? These relationships just use those codes to associate the records, what do you want to remove?

Reply:
opps guess it was defined 

so maybe 

``ruby
ucas_course_code { course.course_code }
provider_code { course.provider.provider_code }
```

since 
```ruby

course
```

as the driver

Reply:
If I understand this correctly, and by "this" I mean "factories", by doing `provider { course }` we ensure that we just re-use the course's provider, instead of creating a separate one.

Reply:
Had a convo with @defong IRL, and he is now happy with this.

Reply:
Ok great.